### PR TITLE
api/types: hostconfig: add some constants/enums and minor code cleanup

### PR DESCRIPTION
**Pull request by thaJeztah**
see individual commits for details

### CODE REVIEW

File: api/types/container/host_config.go [hunk 1]
```diff
@@ -37,60 +44,69 @@ func (c CgroupnsMode) Valid() bool {
...
+	IsolationProcess Isolation = "process" // IsolationProcess is process isolation mode
+	IsolationHyperV  Isolation = "hyperv"  // IsolationHyperV is HyperV isolation mode
+)
+
 // IsDefault indicates the default isolation technology of a container. On Linux this
 // is the native driver. On Windows, this is a Windows Server Container.
 func (i Isolation) IsDefault() bool {
-	return strings.ToLower(string(i)) == "default" || string(i) == ""
+	// TODO consider making isolation-mode strict (case-sensitive)
+	v := Isolation(strings.ToLower(string(i)))
```
Comment (thaJeztah) [inline api/types/container/host_config.go]: Added TODO's for these; I'm not sure if we want these to be case-insensitive (which is different from other options)

**Comment by thaJeztah** [comment]
marking as WIP; I split this from a branch I was working on, think I missed some changes to include here

**Comment by thaJeztah** [comment]
ok, this should be ready for review

@kolyshkin @cpuguy83 PTAL

**Comment by thaJeztah** [comment]
ping @kolyshkin @cpuguy83 PTAL

**Comment by thaJeztah** [comment]
rebased; PTAL

**Comment by thaJeztah** [comment]
Hm.. looks like I broke something;

```

[2020-04-02T23:32:30.310Z] # github.com/docker/docker/daemon/logger/loggerutils/cache
[2020-04-02T23:32:30.310Z] daemon/logger/loggerutils/cache/local_cache.go:39:25: invalid operation: info.Config["mode"] == container.LogModeUnset (mismatched types string and container.LogMode)
[2020-04-02T23:32:30.310Z] # github.com/docker/docker/daemon/logger/loggerutils/cache
[2020-04-02T23:32:30.310Z] daemon/logger/loggerutils/cache/local_cache.go:39:25: invalid operation: info.Config["mode"] == container.LogModeUnset (mismatched types string and container.LogMode)
[2020-04-02T23:33:02.369Z] Building test suite binary /go/src/github.com/docker/docker/integration/config/test.main
[2020-04-02T23:33:04.894Z] Building test suite binary /go/src/github.com/docker/docker/integration/container/test.main
[2020-04-02T23:33:12.995Z] Building test suite binary /go/src/github.com/docker/docker/integration/image/test.main
[2020-04-02T23:33:34.906Z] # github.com/docker/docker/daemon/logger/loggerutils/cache
[2020-04-02T23:33:34.906Z] daemon/logger/loggerutils/cache/local_cache.go:39:25: invalid operation: info.Config["mode"] == container.LogModeUnset (mismatched types string and container.LogMode)
```

**Comment by thaJeztah** [comment]
ping @kolyshkin @cpuguy83 @AkihiroSuda PTAL

File: api/types/container/host_config.go [hunk 2]
```diff
@@ -37,60 +44,69 @@ func (c CgroupnsMode) Valid() bool {
...
-	IsolationDefault = Isolation("default")
-	// IsolationProcess is process isolation mode
-	IsolationProcess = Isolation("process")
-	// IsolationHyperV is HyperV isolation mode
-	IsolationHyperV = Isolation("hyperv")
+	ipcModeNone      = "none"
+	ipcModeHost      = "host"
+	ipcModeContainer = "container"
+	ipcModePrivate   = "private"
+	ipcModeShareable = "shareable"
```
Comment (AkihiroSuda) [inline api/types/container/host_config.go:81]: Can we expose these constants?

**Comment by thaJeztah** [comment]
ping @cpuguy83 @AkihiroSuda @tianon PTAL

**Review by tianon — APPROVED** [review]

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**